llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp

```diff
@@ -2420,29 +2420,27 @@ bool SchedGroup::canAddMI(const MachineInstr &MI) const {
     Result = true;
 
   else if (((SGMask & SchedGroupMask::VMEM) != SchedGroupMask::NONE) &&
-           (TII->isVMEM(MI) || (TII->isFLAT(MI) && !TII->isDS(MI))))
+           TII->isVMEM(MI))
```

Comment:
Am I right in thinking that the isDS test was redundant, since no instructions are both FLAT and DS? @kerbowa

Reply:
Otherwise, it should check for `!isLDSDMA` if we specifically want to avoid DS accesses, I think